File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.size();
        int r=n-1;
        while(r>=0){
            if(s[r] != ' '){
                break;
            }
            r--;
        }
        int l=r;
        while(r>=0){
            if(s[r] == ' '){
                return l-r;
            }
            r--;
        }
        return l-r;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int n = s.length();
        int r = n-1;
        while(r>=0){
            if(s.charAt(r) != ' '){
                break;
            }
            r--;
        }
        int l=r;
        while(r>=0){
            if(s.charAt(r) == ' '){
                return l-r;
            }
            r--;
        }
        return l-r;
    }
}